src/components/Header/Header.tsx

```diff
@@ -0,0 +1,29 @@
+import { RefObject } from 'react';
+
+interface Props {
+  title: RefObject<HTMLInputElement>;
```

Comment:
The name `title` is confusing.
Moreover, in React 18 or earlier we should use `forwardRef` for passing a ref to child component.